src/str.rs

```diff
@@ -241,6 +241,52 @@ macro_rules! take_till_s (
   );
 );
 
+/// `take_until_and_consume_s!(&str) => &str -> IResult<&str, &str>`
+/// generates a parser consuming all chars until the specified string is found and consumes it
+#[macro_export]
+macro_rules! take_until_and_consume_s (
+  ($input:expr, $substr:expr) => (
+    {
+      #[inline(always)]
+      fn shift_window_and_cmp(window: & mut Vec<char>, c: char, substr_vec: &Vec<char>) -> bool {
+        window.push(c);
+        if window.len() > substr_vec.len() {
+          window.remove(0);
+        }
+        window == substr_vec
+      }
+      let res: $crate::IResult<&str, &str> = if $substr.len() > $input.len() {
+        $crate::IResult::Incomplete($crate::Needed::Size($substr.len()))
+      } else {
+        let substr_vec: Vec<char> = $substr.chars().collect();
```

Comment:
I tried `use ::std::vec::Vec;` and `use ::std::vec::Vec as Vec;` and they both warn of unused imports. Should I just make all references to Vec say `::std::vec::Vec` instead of `Vec`?